Implement irq_offload() on Cortex M0

tyrantlizard77 opened the pull request.

Commit:
bbc_microbit: fix 'make debugserver'

Signed-off-by: Andrew Boie <andrew.p.boie@intel.com>

Commit:
arm: implement __svc on Cortex M0

This is needed for irq_offload() and k_oops()/k_panic()

Issue: ZEP-2221
Signed-off-by: Andrew Boie <andrew.p.boie@intel.com>

lpereira commented on arch/arm/core/irq_offload.c:
```diff
@@ -22,9 +22,17 @@ void _irq_do_offload(void)
 void irq_offload(irq_offload_routine_t routine, void *parameter)
 {
-	int key;
+#if defined(CONFIG_ARMV6_M) && defined (CONFIG_ASSERT)
```
Minor nit, but space after the second `defined`.

nashif reviewed (dismissed):
tested on multiple boards, LGTM

MaureenHelm commented on arch/arm/core/swap.S:
```diff
@@ -217,6 +220,17 @@ _thread_irq_disabled:
 #if defined(CONFIG_ARMV6_M)
+#ifdef CONFIG_IRQ_OFFLOAD
+SECTION_FUNC(TEXT, __svc)
+    push {lr}
```
Shouldn't we parse the service call number first? Otherwise this will break if we hit a kernel_oops

andrewboie replied:
That's right, we use SVC for more than one thing now. Will fix.

andrewboie replied:
should be done now

Commit:
arm: fix k_oops on armv6 with interrupts locked

Calling 'svc' on ARMv6 causes a hard fault if interrups are locked.
Force them unlocked before making the svc call.

Signed-off-by: Andrew Boie <andrew.p.boie@intel.com>
